design/accepted/gc_plus/gc-plus-milestone2-execution-plan.md

```diff
@@ -0,0 +1,50 @@
+# Uncommitted Garbage Collection - Milestone 2 Execution Plan
+
+Uncommitted Garbage Collection [Proposal](https://github.com/treeverse/lakeFS/blob/master/design/accepted/gc_plus/uncommitted-gc.md)
+
+## Milestone 1
+The first beta version that was released included:
+1. Implementation of the clean run flow for old and new repository structures (without optimizations)
+2. Mark & Sweep
+3. Integration tests
+4. Backup & Restore
+
+## Milestone 2
+
+### Goals
+1. Removing the limitation of a read-only lakeFS during the job run
+2. Performance improvements - better parallelization of the storage namespace traversal
+3. Implementing Optimized Run
+
+### Non-Goals
+1. Support for non-S3 repositories
+2. Incorporation of committed & uncommitted GC into a single job
+    * Including GC changes, configuration, and behavior changes to fit GC
+3. Metrics and Logging additions
+4. Deployment to lakeFS Cloud
+
+### Plan
+
+* marks dependency
+
+1. Required changes by lakeFS:
+    * [[Get/Link]PhysicalAddress](https://github.com/treeverse/lakeFS/issues/4476)
+        * [Validation of cutoff time](https://github.com/treeverse/lakeFS/issues/4695)
+    * [StageObject API](https://github.com/treeverse/lakeFS/issues/4480)
+    * [CopyObject API](https://github.com/treeverse/lakeFS/issues/4477)
+        * [S3 Gateway CopyObject](https://github.com/treeverse/lakeFS/issues/4478)
+        * [lakeFSFS renameObject method](https://github.com/treeverse/lakeFS/issues/4479)
+    * [Track copied objects in ref-store](https://github.com/treeverse/lakeFS/issues/4562)
+
+3. Performance improvements:
+    * [Optimaized listing on old repository structure](https://github.com/treeverse/lakeFS/issues/4620)
+    * [Efficient listing on committed entries](https://github.com/treeverse/lakeFS/issues/4600)
+    * [Optimized listing on slices](https://github.com/treeverse/lakeFS/issues/4614)
```

Comment:
Thought we already doing this one

Reply:
@nopcoder  Yes you did :) I think we can close this issue

Reply:
removed